commands/operator-sdk/cmd/test/local.go

```diff
@@ -176,3 +189,69 @@ func combineManifests(base, manifest []byte) []byte {
 	}
 	return base
 }
+
+// TODO: add support for multiple deployments and containers (user would have to
+// provide extra information in that case)
+
+// replaceImage searches for a deployment and replaces the image in the container
+// to the one specified in the function call. The function will fail if the
+// number of deployments is not equal to one or if the deployment has multiple
+// containers
+func replaceImage(manifestPath, image string) error {
+	yamlFile, err := ioutil.ReadFile(manifestPath)
+	if err != nil {
+		return err
+	}
+	foundDeployment := false
+	newManifest := []byte{}
+	yamlSplit := bytes.Split(yamlFile, []byte("\n---\n"))
+	for _, yamlSpec := range yamlSplit {
+		if string(yamlSpec) == "" {
+			continue
+		}
+		decoded := make(map[string]interface{})
+		err = yaml.Unmarshal(yamlSpec, &decoded)
+		if err != nil {
+			return err
+		}
+		kind, ok := decoded["kind"].(string)
+		if !ok {
```

Comment:
This and the `kind != "Deployment"` branch can be combined since they're doing the same thing.